src/plugins/data/public/query/query_string/query_history.ts

```diff
@@ -4,60 +4,82 @@
  */
 
 import { BehaviorSubject } from 'rxjs';
-import { DataStorage, Dataset } from '../../../common';
+import { DataStorage } from '../../../common';
 import { Query, TimeRange } from '../..';
 
-// Todo: Implement a more advanced QueryHistory class when needed for recent query history
+const MAX_HISTORY_SIZE = 500;
+export const HISTORY_KEY_PREFIX = 'query_';
+
 export class QueryHistory {
-  constructor(private readonly storage: DataStorage) {}
+  private changeEmitter: BehaviorSubject<any[]>;
```

Comment:
i think the type is just
```ts
    interface QueryHistoryItem {
      time: number;
      query: Query;
      dateRange?: TimeRange;
    };
```

Reply:
imo i think the type should extend savedquery